### PR TITLE
Favourite recipes and Caching of offers

### PR DESCRIPTION
**Pull request by Jaffenheimer**
Favourite recipes can now be added to a user and furthermore be retrieved when desired

### CODE REVIEW

**Comment by Jaffenheimer** [comment]
I have merged caching of ingredients into this since they i thought it would perhaps give some conflicts, and i would just have made it on this branch but accidentally made a new one, so just a NOTE: that caching of Offers (prices etc.) is also in this PR

File: P7Internet.Persistence/FavouriteRecipeRepository/FavouriteRecipeRepository.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+using Dapper;
+using P7Internet.Persistence.Connection;
+using P7Internet.Persistence.RecipeCacheRepository;
+using SharedObjects;
+
+namespace P7Internet.Persistence.FavouriteRecipeRepository;
+
+public class FavouriteRecipeRepository : IFavouriteRecipeRepository
+{
+    private static readonly string TableName = "CachedRecipes";
```
Comment (JJTM20) [inline P7Internet.Persistence/FavouriteRecipeRepository/FavouriteRecipeRepository.cs]: Dette burde være FavouriteRecipesTable?

File: P7Internet.Persistence/FavouriteRecipeRepository/FavouriteRecipeRepository.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        var result = await _cachedRecipeRepository.GetListOfRecipes(guids.ToList());
+        
+        if(result != null)
+            return result;
+        return null;
+    }
+
+    public async Task<bool> Upsert(Guid userId, Guid recipeId)
+    {
+        var query = $@"INSERT INTO {TableName} (UserId, RecipeId)
```
Comment (JJTM20) [inline P7Internet.Persistence/FavouriteRecipeRepository/FavouriteRecipeRepository.cs:51]: Tjek om RecipeId eksisterer i DB

**Review by JJTM20 — APPROVED** [review]

File: P7Internet.Persistence/FavouriteRecipeRepository/FavouriteRecipeRepository.cs (hunk 1 above)
Comment (Jaffenheimer) [reply P7Internet.Persistence/FavouriteRecipeRepository/FavouriteRecipeRepository.cs]: Ændret

File: P7Internet.Persistence/FavouriteRecipeRepository/FavouriteRecipeRepository.cs (hunk 2 above)
Comment (Jaffenheimer) [reply P7Internet.Persistence/FavouriteRecipeRepository/FavouriteRecipeRepository.cs:51]: Done

**Pull request merged**